packages/amplify-graphql-schema-generator/src/generate.ts

```diff
@@ -0,0 +1,53 @@
+import * as os from 'os';
+import { ImportedRDSType } from '@aws-amplify/graphql-transformer-core';
+import { DocumentNode } from 'graphql';
+import { Schema, Engine } from './schema-representation';
+import { generateGraphQLSchema } from './schema-generator';
+import { constructRDSGlobalAmplifyInput } from './input';
+import { MySQLStringDataSourceAdapter } from './datasource-adapter';
+
+const buildSchemaFromString = (stringSchema: string, engineType: ImportedRDSType): Schema => {
+  let schema;
+  let adapter;
+  switch (engineType) {
+    case ImportedRDSType.MYSQL:
+      adapter = new MySQLStringDataSourceAdapter(stringSchema);
+      schema = new Schema(new Engine('MySQL'));
+      break;
+    case ImportedRDSType.POSTGRESQL:
+      // todo change to postgres
+      adapter = new MySQLStringDataSourceAdapter(stringSchema);
+      schema = new Schema(new Engine('Postgres'));
+      break;
+    default:
+      throw new Error('Only MySQL and Postgres Data Sources are supported');
+  }
+
+  const models = adapter.getModels();
+  models.forEach((m) => schema.addModel(m));
+  return schema;
+};
+
+export const renderSchema = (
+  schema: Schema,
+  transformerVersion: number,
+  databaseConfig: any,
+  existingSchema: DocumentNode = undefined,
+): string => {
+  return (
+    constructRDSGlobalAmplifyInput(transformerVersion, databaseConfig, existingSchema) +
+    os.EOL +
+    os.EOL +
+    generateGraphQLSchema(schema, existingSchema)
+  );
+};
+
+export const graphqlSchemaFromRDSSchema = (sqlSchema: string, engineType: ImportedRDSType): string => {
+  const schema = buildSchemaFromString(sqlSchema, engineType);
+
+  const transformerVersion = 2;
```

Comment:
I'm surprised we don't have a centralized place to store transformer versions. How will we remember to catch this when we update the version at some point in future?

Reply:
An alternative here would be to change the function signature to `graphqlSchemaFromRDSSchema = (sqlSchema: string, engineType: ImportedRDSType, includeAuthRule: boolean = false)`

The transformer version is only used here: https://github.com/aws-amplify/amplify-category-api/pull/2004/files#diff-fbea81850cb6d8d9f8b12d224eaebab2b6377705a8799dcc1e7aa9de03537183R25

Reply:
I think I'll actually change to this. I think it makes more sense

Reply:
Looks like that would also require you to remove the transformerversion check from that block though, which may have unintended consequences later if we change the schema structure & associated version. I don't object to passing the version, per se, I am concerned by hardcoding it at arbitrary places through the code, rather than centralizing known versions into a single place that we can refer to throughout the codebase. If we don't already have such a centralization, lets defer the work to create it rather than bundling it into this change.

Reply:
Ok, I'll revert to when transformerVersion is hard coded so that it leaves the function as minimal as possible.